pkg/commands/root.go

```diff
@@ -30,26 +30,24 @@ import (
 // AddCommands adds all subcommands to the root command.
 func AddCommands(ctx context.Context, args []string) *cobra.Command {
 	rootCmd := &cobra.Command{
-		Use:   "bundler",
-		Short: "bundler is tool for inspecting, validation, and modifying components packages and component sets.",
+		Use:   "bundlectl",
+		Short: "bundlectl is tool for inspecting, validation, and modifying components packages and component sets.",
 		Run: func(cmd *cobra.Command, args []string) {
 			cmd.Help()
 		},
 	}
 
-	// TODO(kashomon): Should the GlobalOptionsValues be de-globalized? It's
-	// certainly possible.
 	rootCmd.PersistentFlags().StringVarP(
-		&cmdlib.GlobalOptionsValues.BundleFile, "bundle-file", "f", "", "The path to a bundle file")
+		&cmdlib.GlobalOptionsValues.InputFile, "input-file", "f", "", "The path to an input file")
 
 	rootCmd.PersistentFlags().StringVarP(
-		&cmdlib.GlobalOptionsValues.InputFormat, "in-format", "", "yaml", "The input file format. One of either 'json' or 'yaml'")
+		&cmdlib.GlobalOptionsValues.InputFormat, "in-format", "", "yaml", "The input file format. One of either 'json' or 'yaml'. If an input-file is specified, it is inferred from the file extension.")
 
 	rootCmd.PersistentFlags().StringVarP(
 		&cmdlib.GlobalOptionsValues.OutputFile, "output-file", "o", "", "The path for any output file")
 
 	rootCmd.PersistentFlags().StringVarP(
-		&cmdlib.GlobalOptionsValues.OutputFormat, "format", "", "yaml", "The output file format. One of either 'json' or 'yaml'")
+		&cmdlib.GlobalOptionsValues.OutputFormat, "format", "", "yaml", "The output file format. One of either 'json' or 'yaml'. If an output-file is specified, it is inferred from the file extension.")
```

Comment:
Ah - I see the flag precedence issue now.  If it's important, the approach I normally use is either to make the default decision late (at the point of use), or alternatively to infer the OutputFormat from the filename asap.

Reply:
Done. Went with default decision late.